### PR TITLE
Add 1.74 release notes

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @cuviper @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+<a id="1.74.0-Language"></a>
+
+Language
+--------
+
+- [Codify that `std::mem::Discriminant<T>` does not depend on any lifetimes in T](https://github.com/rust-lang/rust/pull/104299/)
+- [Replace `private_in_public` lint with `private_interfaces` and `private_bounds` per RFC 2145](https://github.com/rust-lang/rust/pull/113126/)
+  Read more in [RFC 2145](https://rust-lang.github.io/rfcs/2145-type-privacy.html).
+- [Allow explicit `#[repr(Rust)]`](https://github.com/rust-lang/rust/pull/114201/)
+- [closure field capturing: don't depend on alignment of packed fields](https://github.com/rust-lang/rust/pull/115315/)
```
Comment (slanterns) [inline RELEASES.md:13]:
```suggestion
- [closure field capturing: don't depend on alignment of packed fields](https://github.com/rust-lang/rust/pull/115315/)
- [Stabilize `impl_trait_projections`](https://github.com/rust-lang/rust/pull/115659)
```

Comment (cuviper) [reply RELEASES.md:13]: How did that miss getting a milestone? (now fixed)

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,87 @@
...
+
+<a id="1.74.0-Internal-Changes"></a>
+
+Internal Changes
+----------------
+
+These changes do not affect any public interfaces of Rust, but they represent
+significant improvements to the performance or internals of rustc and related
+tools.
+
```
Comment (pthariensflame) [inline RELEASES.md:77]: Many of these seem to be actually user-facing?  Mostly in `rustdoc`.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,87 @@
...
+- [Implement `From<[T; N]>` for `Rc<[T]>` and `Arc<[T]>`](https://github.com/rust-lang/rust/pull/114041/)
+- [`impl TryFrom<char> for u16`](https://github.com/rust-lang/rust/pull/114065/)
+- [Stabilize `io_error_other` feature](https://github.com/rust-lang/rust/pull/115453/)
+- [Stabilize the `Saturating` type](https://github.com/rust-lang/rust/pull/115477/)
+- [Stabilize const_transmute_copy](https://github.com/rust-lang/rust/pull/115520/)
+
+<a id="1.74.0-Stabilized-APIs"></a>
+
+Stabilized APIs
+---------------
```
Comment (cuviper) [inline RELEASES.md:42]: @Mark-Simulacrum have you looked at this section yet?

Comment (Mark-Simulacrum) [reply RELEASES.md:42]: Oh, I forgot this wasn't done yet. I'll try to take a look by Monday (hopefully sooner).

**Comment by Mark-Simulacrum** [comment]
Updated with stabilizations and resolved comments.

@bors r+ rollup

File: RELEASES.md (hunk 1 above)
Comment (cuviper) [reply RELEASES.md:13]: @Mark-Simulacrum your last force-push appears to have lost this addition.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,87 @@
...
+
+<a id="1.74.0-Compatibility-Notes"></a>
+
+Compatibility Notes
+-------------------
+
+- [Raise minimum supported Apple OS versions](https://github.com/rust-lang/rust/pull/104385/)
+- [make Cell::swap panic if the Cells partially overlap](https://github.com/rust-lang/rust/pull/114795/)
+- [Reject invalid crate names in `--extern`](https://github.com/rust-lang/rust/pull/116001/)
+
```
[the remark this answers is not in the document]
Comment (cuviper) [reply RELEASES.md:104]: @Mark-Simulacrum this too was lost in the force push.

[the remark this answers is not in the document]
Comment (slanterns) [reply RELEASES.md:104]: I opened a PR to add them back.

**Pull request merged**